Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift

```diff
@@ -2462,37 +2462,80 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
 
   override func visit(_ node: StringSegmentSyntax) -> SyntaxVisitorContinueKind {
     // Looks up the correct break kind based on prior context.
+    func stringLiteralParent() -> StringLiteralExprSyntax? {
+      node.parent?
+        .as(StringLiteralSegmentListSyntax.self)?
+        .parent?
+        .as(StringLiteralExprSyntax.self)
+    }
     func breakKind() -> BreakKind {
-      if let stringLiteralSegments = node.parent?.as(StringLiteralSegmentListSyntax.self),
-        let stringLiteralExpr = stringLiteralSegments.parent?.as(StringLiteralExprSyntax.self)
-      {
+      if let stringLiteralExpr = stringLiteralParent() {
         return pendingMultilineStringBreakKinds[stringLiteralExpr, default: .same]
       } else {
         return .same
       }
     }
 
+    let isMultiLineString = stringLiteralParent()?.openingQuote.tokenKind == .multilineStringQuote
+    // Replace each instance of consecutive whitespace in a string segment with an `.escaped` break.
+    func emitMultilineSegmentTextTokens(segment: Substring) {
+      var currentWord = [Character]()
```

Comment:
Can we build up `Unicode.Scalar`s instead of `Character`s here? That should produce the same result but without incurring the performance penalty of doing grapheme clustering.